lib/python/pyflyby/_interactive.py

```diff
@@ -1814,7 +1817,26 @@ def _enable_reset_hook(self, ip):
         # function to get called twice per cell.  This seems like an
         # unintentional repeated call in IPython itself.  This is harmless for
         # us, since doing an extra reset shouldn't hurt.
-        if hasattr(ip, "input_transformer_manager"):
+        if hasattr(ip, "input_transformers_post"):
+            # In IPython 7.0+, the input transformer API changed.
+            def reset_auto_importer_state(line):
+                # There is a bug in IPython that causes the transformer to be
+                # called multiple times
+                # (https://github.com/ipython/ipython/issues/11714). Until it
+                # is fixed, workaround it by skipping one of the calls.
+                stack = inspect.stack()
+                if any([False,
+                        stack[3].function == 'run_cell_async',
+                        # stack[3].function == 'should_run_async',
+                        # stack[1].function == 'check_complete'
+                ]):
```

Comment:
isn't this equivalent to just `if stack[3].function == 'run_cell_async':`

Reply:
Yea, I wanted to leave the other ones in there as comments, in case this isn't actually correct (as far as I can tell from the tests, it should be). I don't fully understand where the three different places are called. I originally thought the inverse of this was correct, but apparently it isn't.

Reply:
I think it is a little confusing to keep this as it is now.  Can you please add a comment saying why the others are commented out and when to uncomment them.  Also, you probably don't need the `False` in there.

Reply:
I think I added the `False` so that I could comment all three out (which is stupid because `any([])` is the same as `any([False])`).